### DISCUSSION TITLE
Unable to install brew on my new mac

### DISCUSSION THREAD

**Discussion by namlegief** [Homebrew/discussions#2899] [Getting started]
evgene@Evgenes-MacBook-Pro ~ % /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
==> Checking for `sudo` access (which may request your password)...
Password:
==> This script will install:
/opt/homebrew/bin/brew
/opt/homebrew/share/doc/homebrew
/opt/homebrew/share/man/man1/brew.1
/opt/homebrew/share/zsh/site-functions/_brew
/opt/homebrew/etc/bash_completion.d/brew
/opt/homebrew

Press RETURN to continue or any other key to abort:
==> /usr/bin/sudo /usr/sbin/chown -R evgenefeigelman:admin /opt/homebrew
==> Downloading and installing Homebrew...
HEAD is now at d36b61d74 Merge pull request #12824 from MikeMcQuaid/reinstall_cask
error: Not a valid ref: refs/remotes/origin/master
fatal: ambiguous argument 'refs/remotes/origin/master': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
May be I missed something ?

**Comment by carlocab** [comment]
What's the output of
```
ls -la /opt/homebrew
ls -la/opt/homebrew/bin
```
?

**Comment by namlegief** [comment]
```
evgene@Evgenes-MacBook-Pro ~ % ls -la /opt/homebrew

total 104
drwxr-xr-x  32 evgene  admin  1024 Feb  1 18:30 .
drwxr-xr-x   3 root             wheel    96 Feb  1 14:47 ..
-rw-r--r--   1 evgene  admin    51 Feb  1 18:30 .dockerignore
-rw-r--r--   1 evgene  admin   467 Feb  1 18:30 .editorconfig
drwxr-xr-x  15 evgene  admin   480 Feb  1 18:34 .git
drwxr-xr-x   9 evgene  admin   288 Feb  1 18:30 .github
-rw-r--r--   1 evgene  admin  7947 Feb  1 18:30 .gitignore
-rw-r--r--   1 evgene  admin  1342 Feb  1 18:30 .shellcheckrc
drwxr-xr-x   3 evgene  admin    96 Feb  1 18:30 .sublime
-rw-r--r--   1 evgene  admin    65 Feb  1 18:30 .vale.ini
drwxr-xr-x   4 evgene  admin   128 Feb  1 18:30 .vscode
-rw-r--r--   1 evgene  admin   207 Feb  1 18:30 CHANGELOG.md
-rw-r--r--   1 evgene  admin   829 Feb  1 18:30 CONTRIBUTING.md
drwxrwxr-x   2 evgene  admin    64 Feb  1 14:47 Caskroom
drwxrwxr-x   2 evgene  admin    64 Feb  1 14:47 Cellar
-rw-r--r--   1 evgene  admin  1751 Feb  1 18:30 Dockerfile
drwxrwxr-x   2 evgene  admin    64 Feb  1 14:47 Frameworks
-rw-r--r--   1 evgene  admin  1334 Feb  1 18:30 LICENSE.txt
drwxr-xr-x   8 evgene  admin   256 Feb  1 18:30 Library
-rw-r--r--   1 evgene  admin  8069 Feb  1 18:30 README.md
-rw-r--r--   1 evgene  admin   705 Feb  1 18:30 SECURITY.md
drwxrwxr-x   3 evgene  admin    96 Feb  1 18:30 bin
drwxr-xr-x   7 evgene  admin   224 Feb  1 18:30 completions
drwxr-xr-x  70 evgene  admin  2240 Feb  1 18:30 docs
drwxrwxr-x   2 evgene  admin    64 Feb  1 14:47 etc
drwxrwxr-x   2 evgene  admin    64 Feb  1 14:47 include
drwxrwxr-x   2 evgene  admin    64 Feb  1 14:47 lib
drwxr-xr-x   4 evgene  admin   128 Feb  1 18:30 manpages
drwxrwxr-x   2 evgene  admin    64 Feb  1 14:47 opt
drwxrwxr-x   2 evgene  admin    64 Feb  1 14:47 sbin
drwxrwxr-x   3 evgene  admin    96 Feb  1 14:47 share
drwxrwxr-x   3 evgene  admin    96 Feb  1 14:47 var
evgene@Evgenes-MacBook-Pro ~ % ls -la /opt/homebrew/bin 
total 16
drwxrwxr-x   3 evgene  admin    96 Feb  1 18:30 .
drwxr-xr-x  32 evgene  admin  1024 Feb  1 18:30 ..
-rwxr-xr-x   1 evgene  admin  4377 Feb  1 18:30 brew
```

**Reply by carlocab** [reply]
It looks like you might have a partial Homebrew installation. What's the output of `/opt/homebrew/bin/brew config` and `/opt/homebrew/bin/brew doctor`?

**Reply by namlegief** [reply]
I deleted all from /opt and issued the installer again . It looks good now . Thanks a lot

**Reply by Sudhanshujha10** [reply]
how did you deleted all

**Comment by maestroh1git** [comment]
Im having a sort of similar issue; except it involves git

this is it

amaku@Davids-MacBook-Pro ~ % /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
==> Checking for `sudo` access (which may request your password)...
Password:
==> This script will install:
/opt/homebrew/bin/brew
/opt/homebrew/share/doc/homebrew
/opt/homebrew/share/man/man1/brew.1
/opt/homebrew/share/zsh/site-functions/_brew
/opt/homebrew/etc/bash_completion.d/brew
/opt/homebrew

Press RETURN to continue or any other key to abort:
==> /usr/bin/sudo /usr/sbin/chown -R amaku:admin /opt/homebrew
==> Downloading and installing Homebrew...
fatal: remote error: 
  The unauthenticated git protocol on port 9418 is no longer supported.
Please see https://github.blog/2021-09-01-improving-git-protocol-security-github/ for more information.
Failed during: git fetch --force origin

**Reply by gromgit** [reply]
You probably have an `insteadOf` setting that's rewriting GitHub `https:` URLs to `git:` or `ssh:` form, `git config -l | grep -i insteadof` should show you what it is. Remove that config setting, then run the installer again.

**Reply by maestroh1git** [reply]
How do I remove the config setting?


```
amaku@Davids-MacBook-Pro ~ % git config -l | grep -i insteadof
url.git://github.com/.insteadof=https://github.com/
url.git://.insteadof=https://
url.https://.insteadof=ssh://
url.https://github.com/.insteadof=git@github.com:
url.https://github.com.insteadof=git@github.com:
url.https://github.com/.insteadof=git://github.com/
url.https://github.com/.insteadof=git://github.com/
url.git@github.com:.insteadof=git://github.com/
```

**Reply by gromgit** [reply]
The most likely file to edit would be `~/.gitconfig`, though I'd encourage you to read the FILES section of the git-config(1) man page (`man git-config`), which documents all the possible locations where your `insteadOf` directives could be.

I'd also strongly recommend first removing all the `insteadOf` directives you find, then think carefully about what really needs to be rewritten. To be blunt, that's a mess you have there.

**Reply by carlocab** [reply]
You can try
```
git config --list --show-origin | grep -i insteadof
```

**Comment by mmpor** [comment]
Hack

**Reply by mmpor** [reply]
Hack accanet

**Reply by maestroh1git** [reply]
Hack accanet?

**Reply by mmpor** [reply]
> Hack accanet?
> 
>
